### PR TITLE
String coercion for all scout models

### PR DESCRIPTION
**Pull request by islean**
## Description

This PR adds string coercion from number types by default for all basemodels in the scout_export file. Currently some clean functionality is not able to run due to multiple fields marked as string are actually given as a number type, and since upgrading to Pydantic v2, the coercion must be specifically allowed.

I also added default values to the entries with BeforeValidators intended to set a value if not provided, since this does not seem to coincide with the actual behaviour.

### Fixed

- String coercion by default
- Default values set.
 

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b string-coercion-scout-export -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/apps/scout/test_scout_models.py [hunk 1]
```diff
@@ -118,6 +118,22 @@ def test_convert_other_sex(other_sex_case_output: str):
     assert case_obj.model_dump()["individuals"][0][Pedigree.SEX] == PlinkGender.UNKNOWN
 
 
+def test_validate_rank_score_model_float(other_sex_case_output: str):
+    """Test to validate a case when the is set to 'other'"""
+    cases: list = ReadStream.get_content_from_stream(
+        file_format=FileFormat.JSON, stream=other_sex_case_output
+    )
+    case = cases[0]
+    # GIVEN a case that has a float value as rank_model_version
```
Comment (henrikstranneheim) [inline tests/apps/scout/test_scout_models.py:127]:
```suggestion

    # GIVEN a case that has a float value as rank_model_version
```

**Review by henrikstranneheim — APPROVED** [review]
Nice to not lambda 👍

**Comment by islean** [comment]
Test ran without fail in stage with `cg -l DEBUG clean scout-finished-cases -y --days-old 300` working as expected (though with "Could not find any files ready for cleaning for bundle ..." logged)

File: cg/apps/scout/scout_export.py [hunk 2]
```diff
@@ -47,20 +60,22 @@ class Gene(BaseModel):
...
 class ScoutExportCase(BaseModel):
+    model_config = ConfigDict(coerce_numbers_to_str=True)
     id: str = Field(str, alias="_id")
     analysis_date: datetime
     owner: str
     causatives: Optional[List[str]] = None
     collaborators: List[str] = []
     individuals: List[Individual]
-    genome_build: Annotated[str, BeforeValidator(lambda x: str(x))] = GENOME_BUILD_37
+    genome_build: Annotated[str, BeforeValidator(convert_genome_build)] = GENOME_BUILD_37
```
Comment (seallard) [inline cg/apps/scout/scout_export.py:78]: Is the before validator necessary here?

File: cg/apps/scout/scout_export.py [hunk 3]
```diff
@@ -3,41 +3,54 @@
...
     bam_file: Optional[str] = None
     individual_id: str
     sex: Annotated[
         Literal[PlinkGender.UNKNOWN, PlinkGender.MALE, PlinkGender.FEMALE, Gender.OTHER],
         BeforeValidator(set_gender_if_other),
     ]
     father: Annotated[
-        str, BeforeValidator(lambda x: str(x)), BeforeValidator(set_parent_if_missing)
-    ]
+        str, BeforeValidator(set_parent_if_missing)
```
Comment (seallard) [inline cg/apps/scout/scout_export.py:32]: Is this before validator still necessary?

File: cg/apps/scout/scout_export.py (hunk 2 above)
Comment (islean) [reply cg/apps/scout/scout_export.py:78]: Yes - if we want to consider cases where a dict is provided where genome_build is set to None.

**Review by seallard — APPROVED** [review]
👍

**Comment by islean** [comment]
Deployed 2023-10-10 by IO:

```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
cg, version 51.5.9
θ62° 1d [isak.ohlsson@hasta:~] [P_base] 1m58s $ 
```

**Pull request merged**